dts/bindings/gpio/arduino-header-r3.yaml

```diff
@@ -1,10 +1,52 @@
 # Copyright (c) 2019 Foundries.io
+# Copyright (C) 2019 Peter Bigot Consulting, LLC
 # SPDX-License-Identifier: Apache-2.0
 
-title: ARDUINO GPIO HEADER
+title: Arduino Uno (R3) GPIO Header
 
 description: |
-    This is a representation of GPIO pin nodes exposed as headers on Arduino R3
+    GPIO pins exposed on Arduino Uno (R3) headers.
+
+    The Arduino Uno layout provides four headers, two each along
+    opposite edges of the board.  With conventional rotation these are
+    on the left and right side of the board.
+
+    Proceeding counter-clockwise from the upper left:
+    * An 8-pin Power Supply header (upper left).  No pins on this header
+      are exposed by this binding.
+    * A 6-pin Analog Input header (lower left).  This has analog input
+      signals labeled from A0 at the top through A5 at the bottom.
+    * An 8-pin header (lower right).  This has digital input signals
+      labeled from D0 at the bottom D7 at the top;
+    * A 10-pin header (upper right).  This has six additional digital
+      input signals labelled from D8 at the bottom through D13 towards
+      the top, skipping two pins, then finishing with D14 and D15 at the
+      top.
+
+    This binding provides a nexus mapping for 20 pins where parent pins 0
+    through 5 correspond to A0 through A5, and parent pins 6 through 21
+    correspond to D0 through D15, as depicted below:
+
+                             D15 SCL  21
+                             D14 SDA  20
```

Comment:
I wonder if we should keep SCL/SDA reference if we're giving the other functionalities (TX/RX, MISO/MOSI/SKC...)

Reply:
I did that because the ["definitive guide"](https://www.circuito.io/blog/arduino-uno-pinout/) says there are only 14 digital lines.  Zephyr's header chooses to label the SDA/SCL lines as additional digital signals, which is non-standard.  Also the most common use of Arduino signals in Zephyr is to identify the I2C bus controller, albeit that doesn't work right now (#20330).

I'm open to adding more labels, though I believe they're less universal than the I2C ones.

Reply:
If we're taking the pain of writing a (otherwise good) doc, let's make it complete, and not only valid for "today commonly used".
Having only I2C makes it quite special, while I don't think it is.

>Also the most common use of Arduino signals in Zephyr is to identify the I2C bus controller, albeit that doesn't work right now

There are several shields samples that works in the tree.
Spi and serial are also used (at least once in each case) in shields.

Reply:
OK, I'm going to remove SDA/SCL too.  There are as many as four functions that can be applied to any given pin just on the Arduino Uno, and any vendor shield could choose to expose a fifth.

Reply:
I'm fine with this option.